### DISCUSSION TITLE
some suggestions for improving Fooocus

### DISCUSSION THREAD

**Discussion by darth2602** [lllyasviel/Fooocus#2781] [Ideas]
Hello, as a small user, I have some suggestions for improving Fooocus: 

- The ability to scroll vertically/horizontally and zoom in an Inpaint/outpaint image

- integration of control net functions: Depth, UvMap, Open Pose (hand, face, etc.)

- improved faceswap, which works but doesn't look quite the same to me  

- improved dynamic prompting 

- the ability to enhance/add detail to an image when upscaled
  (as Magnific AI does) 

- the appearance of a small explanatory text when hovering over the various functions with the mouse, and the same thing for styles: a small thumbnail and an explanatory text next to it
(fooocus V2, Foocus Sharp,...etc....)

**Comment by mashb1t** [comment]
> The ability to scroll vertically/horizontally and zoom in an Inpaint/outpaint image

You can already use the browser zoom and vary the brush size to achieve this

> improved dynamic prompting

Please elaborate

> the ability to enhance/add detail to an image when upscaled

This is already the case when using 1.5/2x (non-fast) upscaling
